lib/board.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require_relative 'figures/knight'
+require_relative 'figures/rook'
+require_relative 'figures/bishop'
+require_relative 'figures/king'
+require_relative 'figures/queen'
+require_relative 'figures/pawn'
+
+class Board
+  attr_reader :positions, :figures, :display
+
+  def initialize
+    @display = { white: { queen: '♚', king: '♛', rook: '♜', bishop: '♝', knight: '♞', pawn: '♟' },
+                 black: { queen: '♔', king: '♕', rook: '♖', bishop: '♗', knight: '♘', pawn: '♙' } }
+    @figures = { white: {}, black: {} }
+    @positions = Array.new(8) { Array.new(8, nil) }
```

Comment:
The way in which this class is organised is super convoluted. What is the responsibility of this class? 
`@display` is concerned with rendering of the contents of the board, and your `@figures` and `@positions` hold duplicated data. This should be super clear when looking at the `#prepare_board` method: you assign each figure to both of these variables.

If I were to design this, I would have this:
```
@positions = {
  a1: Rook.new(:white),
  b1: Knight.new(:white),
  c1: Bishop.new(:white),
  d1: Queen.new(:white),
  ...and so on
}
```

Data in your `@display` should be distributed among your figures classes:
```ruby
class Queen
  def to_s
    case color
      when :white then "♔"
      when :black then '♚'
    end
  end
end
```